scripts/globals/abilities/full_circle.lua

```diff
@@ -0,0 +1,39 @@
+-----------------------------------
+-- Ability: Full Circle
+-- Causes your luopan to vanish.
+-- Obtained: Geomancer Level 5
+-- Recast Time: 10 seconds
+-- Refunds some of the MP consumed by the Geocolure spell that created the luopan.
+-- Amount of MP restored varies depending on remaining Luopan HP.
+-----------------------------------
+require("scripts/globals/ability")
+require("scripts/globals/status")
+require("scripts/settings/main")
+require("scripts/globals/msg")
+-----------------------------------
+local ability_object = {}
+
+ability_object.onAbilityCheck = function(player, target, ability)
+    if not player:hasPet() or not player:getPetID() == 75 then
+        return xi.msg.basic.REQUIRE_LUOPAN, 0
+    end
+    return 0,0
+end
+
+ability_object.onUseAbility = function(player, target, ability)
+    local luopan        = player:getPet()
+    local hpp_remaining = luopan:getHPP()
+    local mp_cost       = luopan:getLocalVar("MP_COST")
+    local mp_returned   = (0.5 * mp_cost) * hpp_remaining / 100
+    local hp_returned   = 0
```

Comment:
This declaration doesn't seem necessary, since the conditional both sets and restores a value

Reply:
and this is why morning teo should make afternoon teo double check before approving xD I 100% missed this

Reply:
This should be good now, also used the oppertunity to correct full circle formula for mp and hp returned